### PR TITLE
Feature/mtl 1151

### PR DESCRIPTION
**Pull request by mikemoody-amido**
## Link to JIRA ticket

https://hackney.atlassian.net/browse/MTTL-1150

### *What is the problem we're trying to solve*

Implement processing to the PersonAddedToTenure event

#### _Checklist_

- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### CODE REVIEW

**Review by amidomichael — APPROVED** [review]

**Review by evak6979 — APPROVED** [review]

File: PersonListener/Domain/TenureInformation/TenureType.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+namespace PersonListener.Domain.TenureInformation
+{
+    public class TenureType
+    {
+        public string Code { get; set; }
+
+        public string Description { get; set; }
```
Comment (humulla) [inline PersonListener/Domain/TenureInformation/TenureType.cs:7]: Do we need to add the TenureTypes logic to calculate PersonTenureType?

File: PersonListener/Gateway/TenureInfoApiGateway.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        private const string TenureApiUrl = "TenureApiUrl";
+        private const string TenureApiToken = "TenureApiToken";
+        private readonly static JsonSerializerOptions _jsonOptions = CreateJsonOptions();
+
+        public TenureInfoApiGateway(IHttpClientFactory httpClientFactory, IConfiguration configuration)
+        {
+            _httpClientFactory = httpClientFactory;
+            _getTenureApiRoute = configuration.GetValue<string>(TenureApiUrl)?.TrimEnd('/');
+            if (string.IsNullOrEmpty(_getTenureApiRoute) || !Uri.IsWellFormedUriString(_getTenureApiRoute, UriKind.Absolute))
+                throw new ArgumentException($"Configuration does not contain a setting value for the key {TenureApiUrl}.");
```
[the remark this answers is not in the document]
Comment (mikemoody-amido) [reply PersonListener/Gateway/TenureInfoApiGateway.cs:31]: I'll make a minor change to the exception message.

File: PersonListener/Gateway/TenureInfoApiGateway.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+            _getTenureApiToken = configuration.GetValue<string>(TenureApiToken);
+            if (string.IsNullOrEmpty(_getTenureApiToken))
+                throw new ArgumentException($"Configuration does not contain a setting value for the key {TenureApiToken}.");
+        }
+
+        private static JsonSerializerOptions CreateJsonOptions()
```
[the remark this answers is not in the document]
Comment (mikemoody-amido) [reply PersonListener/Gateway/TenureInfoApiGateway.cs:38]: We have no `Startup.cs` in a listener project, but I get the point. I'll leave it here for now so we can get this ticket done, but put something in the backlog to look at it.

File: PersonListener/Domain/TenureInformation/TenureType.cs (hunk 1 above)
Comment (mikemoody-amido) [reply PersonListener/Domain/TenureInformation/TenureType.cs:7]: No. The `PersonTenureType` is on the Tenure `HouseholdMembers` data. We are retrieving this data and not setting it.
The `PersonTenureType` value we retrieve will be a value within the `PersonType` enum.

**Review by humulla — APPROVED** [review]

**Pull request merged**